src/modal_for_noobs/complete_dashboard.py

```
@@ -708,25 +705,59 @@ def show_gpu_options(enable_gpu):
                 return gr.update(visible=enable_remote)
             
-            def start_link_authentication():
-                """Start OAuth-style link authentication."""
+            def start_link_authentication(auth):
+                """Start OAuth-style link authentication using Modal's token flow."""
+
+                async def _flow():
+                    try:
+                        await token_flow._new_token(source="modal-for-noobs")
+                        auth.update(authenticated=True, method="link")
+                    except Exception as e:
```

Ruff (0.11.9)

Do not catch blind exception: `Exception`

(BLE001)